comms/core/src/noise/crypto_resolver.rs

```diff
@@ -66,9 +66,13 @@ impl CryptoResolver for TariCryptoResolver {
     }
 }
 
-fn noise_kdf(shared_key: &CommsDHKE) -> [u8; 32] {
+fn noise_kdf(shared_key: &CommsDHKE) -> CommsNoiseKey {
     let hasher = DomainSeparatedHasher::<Blake256, CommsCoreHashDomain>::new_with_label("noise.dh");
-    Digest::finalize(hasher.chain(shared_key.as_bytes())).into()
+    let mut comms_noise_kdf = CommsNoiseKey::from([0u8; NOISE_KEY_LEN]);
+    comms_noise_kdf
+        .reveal_mut()
+        .copy_from_slice(hasher.chain(shared_key.as_bytes()).finalize().as_ref());
+    comms_noise_kdf
 }
```

Comment:
You can simplify this and avoid an in-memory copy of the KDF output using `finalize_into`.

```rust
use digest::FixedOutput;
use digest::generic_array::GenericArray;
use tari_utilities::safe_array::SafeArray;

let mut comms_noise_key = CommsNoiseKey::from(SafeArray::default());
DomainSeparatedHasher::<Blake256, CommsCoreHashDomain>::new_with_label("noise.dh")
        .chain(shared_key.as_bytes())
        .finalize_into(GenericArray::from_mut_slice(comms_noise_key.reveal_mut()));
comms_noise_key
```

Reply:
Done in [PR 4994](https://github.com/tari-project/tari/pull/4994).